pkg/driver/clickhouse/cluster_parameters.go

```diff
@@ -0,0 +1,80 @@
+package clickhouse
+
+import (
+	"fmt"
+	"net/url"
+)
+
+const (
+	OnClusterQueryParam    = "on_cluster"
+	ZooPathQueryParam      = "zoo_path"
+	ClusterMacroQueryParam = "cluster_macro"
+	ReplicaMacroQueryParam = "replica_macro"
+)
+
+type ClusterParameters struct {
+	OnCluster bool
+	ZooPath string
+	ClusterMacro string
+	ReplicaMacro string
+}
+
+func ClusterParametersFromURL(u *url.URL) *ClusterParameters {
```

Comment:
Maybe the function name should also start with `Extract`, to be `ExtractClusterParametersFromURL`.

Reply:
sounds good to me